feature/14

erikquinn opened the pull request:
Resolves #14.

Add a toggleable _scope command_ bar, and lays foundation for the Scope, its commands, and its collections of radar targets.

n8rzz commented on src/assets/scripts/client/constants/inputConstants.js:
```diff
@@ -0,0 +1,77 @@
+* @type {Object}
+* @final
+*/
+export const KEY_CODES = {
```
almost threw up a little when I saw these in red up above. I should know better, this definitely belongs here 👍

n8rzz commented on index.html:
```diff
@@ -32,7 +32,7 @@
     <div id="log">
     </div>
-    <input id="command" placeholder="enter command" autofocus="true" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" />
+    <input id="command" placeholder="enter aircraft command" autofocus="true" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" />
```
is this file even used?  our html is generated from the handlebars templates and output to the `public` dir.

erikquinn replied:
no idea actually...

sam-irl replied:
Was playing around with index.html previously, I don't believe it is used. I can't test because I'm away from home atm and my poor tablet throws `ENOMEM` when I try to build.

erikquinn replied:
updated with develop reference the merge of #620 (refactor #619), so this file no longer exists here. :+1:

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -332,104 +268,82 @@ export default class InputController {
         }
     }
 
+    // TODO: This does hardly anything of value. It should probably just be removed to simplify things.
```
This comment worries me 😱 

What _is_ this method doing now, isn't this a generic method for any keyboard event?

erikquinn replied:
Well there were _two_ keydown handlers, one for when we were selected in the command bar and another when we weren't. We can instead just do the same action regardless, and when complete, return the focus to the command bar.

See 154392c2f1c8a248610863a00502c7869d42ce56 for what I've just now done done to try to reorganize as such.

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -332,104 +268,82 @@ export default class InputController {
+        // }
     }
 
+    // TODO: Is this really needed??
```
yes.

this is the method that gives us access to the _current_ value of the command input.  

hmm, I see what you're getting at though, changing how this works would definitely be a new issue.

erikquinn replied:
Yeah it's more of a question of rethinking how, or even if, we store the contents of the text field as a variable. Now I think it's in the UI controller and the prop, and both are used, with multiple places where it's being manually parsed to determine the selected aircraft etc. Would be a new issue.

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -332,104 +268,82 @@ export default class InputController {
-        this.input_parse();
-    }
+        // TODO: Refactor out the prop
+        // using `prop` here so CanvasController knows which aircraft is selected
```
I feel like the `EventBus` would be well suited for that, but that smells like a stinky change to make 🤕

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -446,30 +360,29 @@ export default class InputController {
-
-                if (this.input_run()) {
+            case KEY_CODES.ENTER: {
+                if (this.commandBarContext === COMMAND_CONTEXT.AIRCRAFT) {
```
can we genericize this `case` a little?
```javascript
case KEY_CODES.ENTER {
    this.processCmmand();
    this.selecteAircraft();

    break;
}
```

move the `if/else` to  `.processCommand()` and then get rid of the `else` with an early return.  That makes all of this a whole lot easier to test and easier to read.

erikquinn replied:
Love it! Done.

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -722,51 +563,98 @@ export default class InputController {
+                this.$commandInput.css({ color: 'red' });
+
+                return;
+
```
please remove empty line between `return` and next `case`

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -722,51 +563,98 @@ export default class InputController {
+                this.$commandInput.css({ color: 'white' });
+
+                return;
+
```
please remove empty line between `return` and next `case`

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -722,51 +563,98 @@ export default class InputController {
         }
 
-        return this.processTransmitCommand(commandParser);
+        const [successful, response] = this._scopeModel.runScopeCommand(scopeCommandModel);
```
I'm not sure what `successful` is here?  especially since the next statement sends `!successful` as an argument.  What is this actually doing?

erikquinn replied:
Whether the command was understood and executed correctly. The opposite issue is because the `UiController.ui_log()` function takes arguments `message` and `isWarning`.

I did:
```
const isWarning = !successful;

UiController.ui_log(response, isWarning);
```

That should provide more clarity.

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -722,51 +563,98 @@ export default class InputController {
      */
-    processSystemCommand(commandParser) {
-        switch (commandParser.command) {
+    processSystemCommand(AircraftCommandParser) {
```
this should be lowercase since it is and instance.

erikquinn replied:
GAH AUTOCOMPLETE!

done

n8rzz commented on src/assets/scripts/client/InputController.js:
```diff
@@ -833,18 +721,18 @@ export default class InputController {
      * @return {boolean}
      */
-    processTransmitCommand(commandParser) {
+    processTransmitCommand(AircraftCommandParser) {
```
this should be lowercase since it is an instance

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/aircraft/AircraftController.js:
```diff
@@ -569,7 +591,7 @@ export default class AircraftController {
         const aircraftModel = new AircraftModel(initializationProps, this._navigationLibrary);
 
-        this.addItem(aircraftModel);
+        this._eventBus.trigger(EVENT.ADD_AIRCRAFT, aircraftModel);
```
why is this triggering an event for a call to a local method?  are there side-effects that also need to happen now too?  those should be called out in the doc block for the `addItem` method if that is the case.

erikquinn replied:
Figured it was better to have all functions called from the event bus so it's kinda all happening from one place. I can remove and call locally here and then trigger if that's preferable?

n8rzz replied:
yeah, just all this locally.  The `EventBus` should only be used for cross-class communication.  If you just need to call a local method, do that instead of using the event bus (unless there is another class that is also listening for the event).

erikquinn replied:
> If you just need to call a local method, do that instead of using the event bus (unless there is another class that is also listening for the event).

Well that's what I'm saying though, is that there _is_ another class listening for the event. The Aircraft Controller creates the aircraft then passes the new model to the event bus with the `EVENT.ADD_AIRCRAFT` event. Listeners are the `AircraftCollection` (to add the aircraft) and the `RadarTargetCollection` (to add the radar target).

n8rzz replied:
ok.  could you add a comment to that affect?

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/canvas/CanvasController.js:
```diff
@@ -776,19 +778,22 @@ export default class CanvasController {
      */
-    canvas_draw_aircraft(cc, aircraft) {
+    canvas_draw_radar_target(cc, radarTargetModel) {
+        const aircraftModel = radarTargetModel.aircraftModel;
```
you should use destructuring here:
```javascript
const { aircraftModel } = radarTargetModel;
```

erikquinn replied:
Done.

n8rzz commented on src/assets/scripts/client/canvas/CanvasController.js:
```diff
@@ -1029,36 +1034,47 @@ export default class CanvasController {
+            this.canvas_draw_radar_target(cc, radarTargetModels[i]);
             cc.restore();
         }
+        // for (let i = 0; i < prop.aircraft.list.length; i++) {
```
remove this commented out block if it is deprecated

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/canvas/CanvasController.js:
```diff
@@ -1029,36 +1034,47 @@ export default class CanvasController {
-    canvas_draw_info(cc, aircraft) {
-        if (!aircraft.isVisible() || aircraft.hit) {
+    canvas_draw_data_block(cc, radarTargetModel) {
+        const aircraftModel = radarTargetModel.aircraftModel;
```
this can use destructuring

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/canvas/CanvasController.js:
```diff
@@ -1190,9 +1203,18 @@ export default class CanvasController {
             cc.lineTo(-halfWidth, -halfHeight); // top-left corner
             cc.lineTo(-halfWidth, -point1);  // begin side cutout
-            cc.arc(-halfWidth - barHalfWidth, -lock_offset, lock_size / 2 + barHalfWidth, clipping_mask_angle - pi / 2, 0);
+            cc.arc(-halfWidth - barHalfWidth,
```
these inline maths drive me nuts!  alas, for another day

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+import _first from 'lodash/first';
```
stopping point 

🛏 🕚

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+    COMMAND_FUNCTIONS
+} from './scopeCommandMap';
+
+// FIXME: Remove dummy sector codes
```
I 👀 a `FIXME`

erikquinn replied:
Yeeahhh I guess that'll have to be a todo anyway. Involves also hashing out the SectorModel and SectorCollection.

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+ * @class ScopeCommand
+ */
+export default class ScopeCommand {
+    constructor(command) {
```
missing doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+ */
+export default class ScopeCommand {
+    constructor(command) {
+        this.aircraftReference = '';
```
could you add doc blocks for these props too?

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+        this._init(command);
+    }
+
+    _init(command) {
```
missing doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+        this.commandArguments = this._extractCommandArguments(pieces);
+    }
+
+    _extractCommandArguments(command) {
```
missing doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+        return command.splice(1, command.length - 2);
+    }
+
+    _extractCommandFunction(command) {
```
missing doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+
+    _extractCommandFunction(command) {
+        const firstElement = _first(command);
+        if (_has(EXPLICIT_COMMANDS, firstElement)) {
```
missing empty line above the `if`

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+            return COMMAND_FUNCTIONS.INITIATE_HANDOFF;
+        }
+
+        if (firstElement.indexOf('/') !== -1 || firstElement.length < 2) {
```
what does the `/` mean?  what is it for?

erikquinn replied:
The only implied command a `/` is used with is the move data block command. I tried for a while to figure out how or where to neatly extract it to, but all options were clunky so I just left it there.

n8rzz commented on src/assets/scripts/client/parsers/scopeCommandParser/ScopeCommandModel.js:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    _extractCommandArguments(command) {
+        if (_values(IMPLIED_COMMANDS).indexOf(this.commandFunction) !== -1) {
```
I've not used `_values()` before, is that like `Object.keys()` only for collecting the values of an object?

erikquinn replied:
Right. Can't use `_has()` because that gets the keys, and we need the values.

n8rzz commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+ * @class RadarTargetCollection
+ */
+export default class RadarTargetCollection extends BaseCollection {
+    constructor(theme) {
```
missing doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+    constructor(theme) {
+        super();
+
+        this._eventBus = EventBus;
```
missing doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @property items
+     * @type {array}
+     */
+    get items() {
```
isn't this already defined in the `BaseCollection`?

erikquinn replied:
Negative, it is not. :confused:

n8rzz commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @private
+     * @chainable
+     */
+    _init() {
```
missing param or incorrect doc block

erikquinn replied:
Done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+    }
+
+    /**
+     * Add the provided `RadarTargetModel` instance to the collection
```
stopping point, pt. 2

:man_cook: 🕡

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+     * @for RadarTargetModel
+     * @method disable
+     */
+    disable() {
```
move to after `_init()` with other lifecycle methods defined in calling order

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+     * @for RadarTargetModel
+     * @method enable
+     */
+    enable() {
```
move to after `_init()` with other lifecycle methods defined in calling order.  Is this method even called I don't see it in the constructor?

erikquinn replied:
done. wasn't, but is now 👍

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+    }
+
+    /**
+    * Mark this radar target as being controlled by "our" ScopeModel
```
this comment looks like copy pasta.  it describes another method

erikquinn replied:
good catch. done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+        let desiredDirection = commandArguments;
+        let desiredLength = '';
+
+        // FIXME: Extract this to a constants file!
```
`FIXME`

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+        }
+
+        if (desiredDirection !== '' && !_isNaN(desiredDirection)) {
+            const dataBlockPositionMap = { 8: 360, 9: 45, 6: 90, 3: 135, 2: 180, 1: 225, 4: 270, 7: 315, 5: 'ctr' };
```
this obj should be a constant somewhere

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+     * @for RadarTargetModel
+     * @method reset
+     */
+    reset() {
```
move to after `_init()` with other lifecycle methods defined in calling order

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+     * @for RadarTargetModel
+     * @method _initializeScratchPad
+     */
+    _initializeScratchPad() {
```
move to after `_init()` with other lifecycle methods defined in calling order

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+        this._sectorCollection = [];
+        this._theme = THEME.DEFAULT;
+
+        this.radarTargetCollection = [];
```
- remove empty line above this one
- this should default to `null` since it later becomes a `RadarTargetCollection` instance

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+     * @for ScopeModel
+     * @method disable
+     */
+    disable() {
```
move to after `_init()` with other lifecycle methods defined in calling order

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+     * @for ScopeModel
+     * @method enable
+     */
+    enable() {
```
move to after `_init()` with other lifecycle methods defined in calling order.  does this method ever get called?

erikquinn replied:
done, and fixed

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+import { DECIMAL_RADIX } from '../constants/globalConstants';
+import { THEME } from '../constants/themes';
+
+export default class ScopeModel {
```
missing doc block

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+        this._eventBus = EventBus;
+        this._navigationLibrary = NavigationLibrary;
+        // TODO: Use this!
+        this._sectorCollection = [];
```
doc block and default to `null` if this becomes an instance of a class

erikquinn replied:
done

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+     * @for ScopeModel
+     * @method moveDataBlock
+     * @param radarTargetModel {RadarTargetModel}
+     * @param commandArguments {array}
```
should this be: `array<string>`?

erikquinn replied:
hm. actually no, it's just a string. I reconfigured some stuff and didn't update the docblocks. Done now. 👍

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+     * @for ScopeModel
+     * @method propogateDataBlock
+     * @param radarTargetModel {RadarTargetModel}
+     * @param sectorCode {array} handoff code for the receiving sector
```
should this be: array<string>?

erikquinn replied:
no, string. fixed

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+     * @return result {array} [success of operation, system's response]
+     */
+    runScopeCommand(scopeCommandModel) {
+        const functionName = scopeCommandModel.commandFunction;
```
I'm not a fan of using the words `function` and `arguments` as part of var names

erikquinn replied:
seems appropriate for the particular case, but I know what you mean. Any ideas?

n8rzz replied:
I suppose this is a very minor gripe and I can't really think of anything better at the moment.  

In the interest of moving this along, I'll allow it 😉

n8rzz commented on src/assets/scripts/client/scope/ScopeModel.js:
```diff
@@ -0,0 +1,208 @@
+            return [false, 'ERR: UNKNOWN AIRCRAFT'];
+        }
+
+        return this[functionName](radarTargetModel, ...functionArguments);
```
this is sneaky clever, be sure to add a comment on what is happening here, and why.

erikquinn replied:
done

n8rzz commented on test/aircraft/AircraftModel.spec.js:
```diff
@@ -38,19 +38,19 @@ ava('.matchCallsign() returns false when passed a flightnumber that is not inclu
 ava('.matchCallsign() returns true when passed a flightnumber that is included in #callsign', (t) => {
```
stopping pt. 3

erikquinn commented on src/assets/scripts/client/AppController.js:
```diff
@@ -136,7 +136,8 @@ export default class AppController {
         this.navigationLibrary = new NavigationLibrary(initialAirportData);
         this.airlineController = new AirlineController(airlineList);
-        this.aircraftController = new AircraftController(aircraftTypeDefinitionList, this.airlineController, this.navigationLibrary);
+        this.scopeModel = new ScopeModel();
```
There would be one scope to any player in a session, whether of the `controller` or `pseudopilot` or when acting as both in a single player session.

So there will be multiple instances that interact with each other.

erikquinn commented on src/assets/scripts/client/InputController.js:
```diff
@@ -276,6 +215,9 @@ export default class InputController {
         this.input.isMouseDown = false;
+
+        // TODO: Do something with this event or stop collecting it; it causes lint errors
+        return event;
```
Added #710.

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @for RadarTargetModel
+     * @method disable
+     */
+    disable() {
```
Done

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @for RadarTargetModel
+     * @method disable
+     */
+    enable() {
```
done

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @param aircraftModel {AircraftModel}
+     * @return radarTargetModel {RadarTargetModel}
+     */
+    getRadarTargetModelFromAircraftModel(aircraftModel) {
```
done

erikquinn replied:
also done for other similar methods to match the convention

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+    getRadarTargetModelFromAircraftModel(aircraftModel) {
+        // Store variable because `this` within lodash `_filter` has different scope
+        const radarTargetModels = this._items;
+        const results = _filter(radarTargetModels, (radarTargetModel) =>
```
really? that seems gross 😕

whateva! done :)

n8rzz replied:
it is gross, I agree.  but because you're using a direct return, its preferable to have that all on the same line.  Otherwise some other dev in the future could get confused and forget that it's a direct return (like me 😊)

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @param aircraftReference {string} the CID, squawk code, or callsign assigned to an aircraft
+     * @return radarTargetModel {RadarTargetModel}
+     */
+    getRadarTargetModelFromAircraftReference(aircraftReference) {
```
wait, don't you want this `findRadarTargetModelForAircraftReference()`?

n8rzz replied:
words are hard, yes I did mean that

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+    getRadarTargetModelFromAircraftReference(aircraftReference) {
+        // Store variable because `this` within lodash `_filter` has different scope
+        const radarTargetModels = this._items;
+        const results = _filter(radarTargetModels, (radarTargetModel) =>
```
done

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @param aircraftModel {AircraftModel}
+     */
+    removeRadarTargetModel = (aircraftModel) => {
+        const idToRemove = aircraftModel.id;
```
not sure! removed the excessive verbosity 😉

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+    resetAllRadarTargets() {
+        const radarTargetModels = this._items;
+
+        _forEach(radarTargetModels, (radarTargetModel) => radarTargetModel.reset());
```
noted and done

erikquinn commented on src/assets/scripts/client/scope/RadarTargetCollection.js:
```diff
@@ -0,0 +1,214 @@
+     * @param themeName {string}
+     * @private
+     */
+    _setTheme = (themeName) => {
```
it's all over the GD place. anything that needs to access the theme is storing it locally, and has its own `_setTheme` method which is registered to be called by the event bus on `EVENT.SET_THEME`. Open to suggestions on a new issue on how to improve that system.

n8rzz replied:
ahh yes, I remember discussing that little feature when you implemented it.  I'm sure we'll think of something someday.  for now, though, it does work!

erikquinn commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+ * @class RadarTargetModel
+ */
+export default class RadarTargetModel {
+    constructor(theme, aircraftModel) {
```
done

erikquinn commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+         */
+        this._isUnderOurControl = true;
+
+        // TODO: Store the aircraft's initial route here. Yes, we want to intentionally
```
I aim to please!

erikquinn commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+        this._dataBlockLeaderLength = this._theme.DATA_BLOCK.LEADER_LENGTH;
+        // TODO: This getter doesn't give us what we want. Seems like one does
+        // not exist actually. We want the full route string, including past legs.
+        this._routeString = aircraftModel.fms.currentRoute;
```
AH yes I did in fact miss it. Was called `getFlightPlanRouteForStripView`. Abstracted and redirected to use:
 - `.getFlightPlanRouteString()`
- `.getFlightPlanRouteStringWithDots()` via `.getFlightPlanRouteString()`
- `.getFlightPlanRouteStringWithSpaces()` via `.getFlightPlanRouteString()`

erikquinn commented on src/assets/scripts/client/scope/RadarTargetModel.js:
```diff
@@ -0,0 +1,411 @@
+        // not exist actually. We want the full route string, including past legs.
+        this._routeString = aircraftModel.fms.currentRoute;
+
+        this._initializeScratchPad();
```
Done.

Could we adopt a convention of comments dividing these method sections, eg when collapsed:

```
/**
constructor() {...}

/**
get something() {...}

/**
get somethingElse() {...}

/**
get otherThings() {...}

/**
_init() {...}

/**
_init() {...}

/**
enable() {...}

/**
disable() {...}

/**
reset() {...}

/**
constructor() {...}

/**
constructor() {...}

/**
constructor() {...}

/**
constructor() {...}

/**
constructor() {...}

erikquinn replied:
Done.

Could we adopt a convention of comments dividing these method sections, eg when collapsed:

```
/**
constructor() {...}

/**
get something() {...}

/**
get somethingElse() {...}

/**
get otherThings() {...}

// ------------------------ LIFECYCLE ------------------------

/**
_init() {...}

/**
_init() {...}

/**
enable() {...}

/**
disable() {...}

/**
reset() {...}

// ------------------------ PUBLIC ------------------------

/**
someMethod() {...}

/**
anotherMethod() {...}

/**
lastMethod() {...}

// ------------------------ PRIVATE ------------------------

/**
_privateMethod() {...}

/**
_anotherPrivateMethod() {...}

I'm growing tired of the confusion with order. We need to figure out what the plan is and stick to it; and I am not always clear on what order you want in this regard, and I think dividers like these would instantly make it obvious to everyone, and easier to comply.

n8rzz replied:
I can get behind that.  wanna through that in an issue and we can figure out the details?

erikquinn replied:
Sounds like a plan. Added #712.

erikquinn commented:
Addressed review feedback.

To prevent needless examination of old conversations, here are the unresolved conversations:
- [ ] https://github.com/openscope/openscope/pull/597#discussion_r138789792
- [ ] https://github.com/openscope/openscope/pull/597#discussion_r139055874

n8rzz commented:
@erikquinn Thats a wonderful idea!  as my mother would say: _'Bless your heart!'_

n8rzz commented:
@erikquinn With all the changes you made yesterday, I'd say we can get this merged (after those conflicts are addressed).  If we run into anything we don't like, that can be tackled with a new issue.
